domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2429,7 +2449,34 @@ private FacetManager buildFacetManager(CQLWrapper wrapper, QueryHelper queryHelp
    * @param replyHandler  the result; the JSON is encoded as a String
    */
   public void getByIdAsString(String table, String id, Handler<AsyncResult<String>> replyHandler) {
-    getById(table, id, string -> string, replyHandler);
+    getByIdAsString(null, table, id, replyHandler);
+  }
+
+  /**
+   * Get the jsonb by id and return it as a String. Query will be executed on
+   * <code>conn</code> if provided.
+   * @param conn  if provided, the connection on which to execute the query on.
+   * @param table  the table to search in
+   * @param id  the value of the id field
+   * @param replyHandler  the result; the JSON is encoded as a String
+   */
+  public void getByIdAsString(AsyncResult<SQLConnection> conn,
+      String table, String id, Handler<AsyncResult<String>> replyHandler) {
+    getById(conn, false, table, id, string -> string, replyHandler);
+  }
+
+  /**
+   * Get the jsonb by id and return it as a String. Query will be executed on
+   * <code>conn</code> if provided. Selected row will be locked using
+   * <code>select ... for update<code> clause.
```

Comment:
The key information should be in the beginning:

`Lock the row using <code>select ... for update</code> and return jsonb as String.`